contracts/HATKlerosConnector.sol

```diff
@@ -0,0 +1,481 @@
+        dispute.externalDisputeId = externalDisputeId;
+        externalIDtoLocalID[externalDisputeId] = localDisputeId;
+
+        if (msg.value > arbitrationCost) payable(_disputer).send(msg.value - arbitrationCost);
+
+        emit Challenged(_claimId);
+        emit Dispute(klerosArbitrator, externalDisputeId, metaEvidenceUpdates, localDisputeId);
+        emit Evidence(klerosArbitrator, localDisputeId, _disputer, _evidence);
+    } 
+
+    /** @dev Give a ruling for a dispute. Can only be called by the Kleros arbitrator.
+     *  @param _disputeId ID of the dispute in the Kleros arbitrator contract.
+     *  @param _ruling Ruling given by the arbitrator. Note that 0 is reserved for "Refused to arbitrate".
+     */
+    function rule(uint256 _disputeId, uint256 _ruling) external override {
+        uint256 localDisputeId = externalIDtoLocalID[_disputeId];
+
+        DisputeStruct storage dispute = disputes[localDisputeId];
+
+        require(!dispute.resolved, "Already resolved");
+        require(_ruling <= RULING_OPTIONS, "Invalid ruling option");
+        require(address(klerosArbitrator) == msg.sender, "Only the arbitrator can execute");
+        uint256 finalRuling = _ruling;
+    
+        // If one side paid its fees, the ruling is in its favor. Note that if the other side had also paid, an appeal would have been created.
+        Round storage round = dispute.rounds[dispute.rounds.length - 1];
+        if (round.fundedSides.length == 1) finalRuling = round.fundedSides[0];
+
+        dispute.ruling = Decision(finalRuling);
+        dispute.resolved = true;
+
+        bytes32 claimId = dispute.claimId;
+        if (finalRuling == uint256(Decision.ExecuteResolution)) {            
+            hatArbitrator.executeResolution(dispute.vault, claimId);
+        } else {
+            // Arbitrator dismissed the resolution or refused to arbitrate (gave 0 ruling).
+            hatArbitrator.dismissResolution(dispute.vault, claimId);
+        }
+      
+        emit Ruling(IArbitrator(msg.sender), _disputeId, finalRuling);
+    }
+
+    /** @dev Submit a reference to evidence. EVENT.
+     *  @param _localDisputeId The id of the related dispute.
+     *  @param _evidenceURI Link to evidence.
+     */
+    function submitEvidence(uint256 _localDisputeId, string calldata _evidenceURI) external override {
+        DisputeStruct storage dispute = disputes[_localDisputeId];
+        // Note that by reading dispute's value we also check that it exists.
+        require(!dispute.resolved, "Dispute already resolved");
+
+        emit Evidence(klerosArbitrator, _localDisputeId, msg.sender, _evidenceURI);
+    }
+
+    // ************************ //
+    // *       Appeals        * //
+    // ************************ //
+
+    /** @dev Takes up to the total amount required to fund a side. Reimburses the rest. Creates an appeal if both sides are fully funded.
```

Comment:
```suggestion
    /** @dev Takes up to the total amount required to fund a side. Reimburses the rest. Creates an appeal if two of the three sides are fully funded.
```

Reply:
Shouldn't it only be possible to fund the yes or no options? Then have the neutral option be counted as a no?